managers/templates/managers/project_form.html

```diff
@@ -33,15 +35,14 @@ <h2>
   {% trans 'Do you want to delete' %} <br/>
   <b>{{object.name}}</b> {% trans 'project' %}?
 </div>
+
+<link rel="stylesheet" href="//code.jquery.com/ui/1.12.1/themes/smoothness/jquery-ui.css">
+<script src="//code.jquery.com/ui/1.12.1/jquery-ui.js"></script>
```

Comment:
This is not correct solution for this. 

The main problem is that `{% block extra_script %}{% endblock %}` was removed from the `base.html` template, so it must be added there first, at the bottom, above closing `</body>` tag. It is placed there because all scripts should be loaded after the rest of the html page is rendered. If they are placed in the head, selectors might apply to things that are not rendered yet. Off course they might be also applied just after the HTML code to which they are related, but it is good practice to have them in single place, not scattered around the page.

Second thing is that `<script src="//code.jquery.com/ui/1.12.1/jquery-ui.js"></script>` is missing so please add it to the `{% block extra_script %}{% endblock %}` in `project_form.html` template.

Please also move existing scripts from `<head>` above the `{% block extra_script %}{% endblock %}` in `base.html`.

Missing `{% block extra_script %}{% endblock %}` is also causing datepickers to not work, and I am sure they were working so this block must have been removed accidentally

Reply:
When we had `{% block extra_script %}` at the bottom of the base, the javascript dependent project joining feature crashed on page load making it impossible to use it. Report deletion feature was also disabled due to same error. This was because the scripts on those pages (and I mean the custom .js files themselves) were loaded inside the _<head>_ tag in these views and thus they where crashing, because all of dependencies in base were not yet loaded.

I am familiar with principle of loading the scripts only after the rest of the template is loaded to boost the rendering time. I apologise for resulted mess in 'base.html' due to my interference, but project join feature is essential for employee and this bug was detected on the day of release. We absolutely couldn't have released the app in such state and thus I was required to find another ASAP fix.

If I understand correctly, every single javascript reference in all templates must be moved to corresponding '{% block extra_script %}' below the rest of the code? I will apply the changes shortly and I'll check if all features work as expected.

Reply:
>  apologise for resulted mess in 'base.html' due to my interference

Sure don't worry, this project is too about learning right? :) 

> If I understand correctly, every single javascript reference in all templates must be moved to corresponding {% block extra_script %} below the rest of the code? I will apply the changes shortly and I'll check if all features work as expected.

Yes, if you want you can refactor existing templates in separate PR. All scripts should be moved to this block, while all css'es into `extra_head`.

Reply:
By the way, I believe you've made the typo because the javascript package you said was missing is included in the snippet above. Which package is missing?

Reply:
Also, should there be separate blocks in each template? Or should all *.css* and *.js* imports in entire app be stored in `base.html`?